lib/remap-functions/src/tag_types_externally.rs

```diff
@@ -0,0 +1,229 @@
+use remap::prelude::*;
+use shared::btreemap;
+use std::collections::BTreeMap;
+
+#[derive(Clone, Copy, Debug)]
+pub struct TagTypesExternally;
+
+impl Function for TagTypesExternally {
+    fn identifier(&self) -> &'static str {
+        "tag_types_externally"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            accepts: |_| true,
+            required: true,
+        }]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required("value")?.boxed();
+
+        Ok(Box::new(TagTypesExternallyFn { value }))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct TagTypesExternallyFn {
+    value: Box<dyn Expression>,
+}
+
+impl Expression for TagTypesExternallyFn {
+    fn execute(&self, state: &mut state::Program, object: &mut dyn Object) -> Result<Value> {
+        let value = self.value.execute(state, object)?;
+        let tagged_externally = tag_type_externally(&value);
+
+        Ok(tagged_externally)
+    }
+
+    fn type_def(&self, state: &state::Compiler) -> TypeDef {
+        self.value
+            .type_def(state)
+            .into_fallible(false)
+            .with_constraint(value::Kind::Map | value::Kind::Array)
```

Comment:
Is there a way to say "this is an `Array` when `value` is an `Array`, and a `Map` otherwise"?

Reply:
Do you have a tip here, @JeanMertz?

Reply:
Nothing builder-like right now, no, you'll have to use an if-else branch.